src/ocrd/cli/resmgr.py

```diff
@@ -119,16 +118,33 @@ def download(any_url, no_dynamic, resource_type, path_in_archive, allow_uninstal
                 'path_in_archive': path_in_archive}]
             )]
     for this_executable, this_reslist in reslist:
+        if not location:
+            # TODO @mehmedGIT: What if there is no 'resource_locations' in this executable ocrd tool json?
+            #  Then the location will always default to None
+            location = get_ocrd_tool_json(this_executable)['resource_locations'][0]
+        elif location not in get_ocrd_tool_json(this_executable)['resource_locations']:
+            log.error(f"The selected --location {location} is not in the {this_executable}'s resource search path, "
+                      f"refusing to install to invalid location")
+            sys.exit(1)
+        if location == 'module':
+            basedir = get_moduledir(this_executable)
+            if not basedir:
+                basedir = resmgr.location_to_resource_dir('data')
+        else:
+            # TODO @mehmedGIT: The basedir always defaults to CWD when location is not passed from the CLI,
+            #  this may be an undesired default behaviour. Consider setting that up with a default path somewhere.
+            #  For example, RESOURCES_DIR_SYSTEM or config.XDG_DATA_HOME, for consistency and predictability.
```

Comment:
I would be open to changing the default to `config.XDG_DATA_HOME` but I think there was a reason for this default behavior but cannot remember it now. @bertsky do you?

Reply:
We can still keep the default to `CWD` if `config.XDG_DATA_HOME` was never set. I would still prefer the default to be `RESOURCES_DIR_SYSTEM`, i.e., `/usr/local/share/ocrd-resources`, if `XDG_DATA_HOME` is None.

Reply:
Indeed, as in the previous comment, I overlooked that `type=click.Choice([])` defaults to the first entry.

Reply:
> Based on the above, `location` is set to the first allowable among `resource_locations`, which most often is just data. So I don't understand what you mean by `defaults to CWD`.

This is not true. If no location is provided from the CLI, it defaults to None. Maybe in the past, the first option was click.Choice([]) was considered the default, but that is no longer true unless specified manually with `default='data'`